### PR TITLE
feat(spanner): add numeric samples

### PR DESCRIPTION
**Pull request by hengfengli**

### CODE REVIEW

**Comment by hengfengli** [comment]
The CI test fails due to a random error not related to this change: 

```bash
  1) Error:
Google Cloud Firestore API samples - Distributed Counter#test_0002_get_document:
Google::Cloud::NotFoundError: 5:No document to update: projects/ruby-firestore-ci/databases/(default)/documents/shards/5. debug_error_string:{"created":"@1600233739.544228349","description":"Error received from peer ipv4:173.194.216.95:443","file":"src/core/lib/surface/call.cc","file_line":1062,"grpc_message":"No document to update: projects/ruby-firestore-ci/databases/(default)/documents/shards/5","grpc_status":5}
    /usr/local/bundle/gems/google-cloud-firestore-v1-0.2.0/lib/google/cloud/firestore/v1/firestore/client.rb:818:in `rescue in commit'
    /usr/local/bundle/gems/google-cloud-firestore-v1-0.2.0/lib/google/cloud/firestore/v1/firestore/client.rb:784:in `commit'
    /usr/local/bundle/gems/google-cloud-firestore-2.1.0/lib/google/cloud/firestore/service.rb:131:in `commit'
    /usr/local/bundle/gems/google-cloud-firestore-2.1.0/lib/google/cloud/firestore/batch.rb:403:in `commit'
    /usr/local/bundle/gems/google-cloud-firestore-2.1.0/lib/google/cloud/firestore/client.rb:581:in `batch'
    /usr/local/bundle/gems/google-cloud-firestore-2.1.0/lib/google/cloud/firestore/document_reference.rb:404:in `update'
    /tmpfs/src/github/ruby-docs-samples/firestore/distributed_counters.rb:50:in `increment_counter'
    /tmpfs/src/github/ruby-docs-samples/firestore/acceptance/distributed_counters_test.rb:38:in `block (3 levels) in <top (required)>'
```

File: spanner/spanner_samples.rb [hunk 1]
```diff
@@ -924,6 +971,29 @@ def query_with_parameter project_id:, instance_id:, database_id:
...
+  require "google/cloud/spanner"
+  require "bigdecimal"
+
+  spanner = Google::Cloud::Spanner.new project: project_id
+  client  = spanner.client instance_id, database_id
+
+  sql_query = "SELECT VenueId, Revenue FROM Venues WHERE Revenue < @revenue"
+
+  params      = { revenue: BigDecimal("100000") }
+  param_types = { lastName: :NUMERIC }
```
Comment (thiagotnunes) [inline spanner/spanner_samples.rb:989]: I think there is a typo here, shouldn't it be:

```ruby
params      = { revenue: BigDecimal("100000") }
param_types = { revenue: :NUMERIC } # this should be revenue, not lastName
```

File: spanner/Gemfile.lock [hunk 2]
```diff
@@ -3,11 +3,11 @@ GEM
   specs:
     addressable (2.7.0)
       public_suffix (>= 2.0.2, < 5.0)
-    concurrent-ruby (1.1.6)
```
Comment (thiagotnunes) [inline spanner/Gemfile.lock:6]: nit: should library upgrades be part of the same PR or could we push them out separately?

Comment (hengfengli) [reply spanner/Gemfile.lock:6]: I only upgraded `google-cloud-spanner` to the latest version, which is needed for the code samples. Usually, I do it together and I would not worry too much about it.

File: spanner/spanner_samples.rb (hunk 1 above)
Comment (hengfengli) [reply spanner/spanner_samples.rb:989]: Nice catch.

**Review by thiagotnunes — APPROVED** [review]
LGTM

**Comment by skuruppu** [comment]
@TheRoyalTnetennba if you could help us merge in this one, that would be great. Merging is currently blocked on an unrelated firestore test failure.

**Pull request merged**